builtin/logical/nomad/backend_test.go

```diff
@@ -57,24 +64,61 @@ func prepareTestContainer(t *testing.T) (func(), *Config) {
 		if err != nil {
 			return nil, err
 		}
-		aclbootstrap, _, err := nomad.ACLTokens().Bootstrap(nil)
-		if err != nil {
-			return nil, err
+		if manuallyBootstrap {
+			aclbootstrap, _, err := nomad.ACLTokens().Bootstrap(nil)
+			if err != nil {
+				return nil, err
+			}
+			nomadToken = aclbootstrap.SecretID
+			t.Logf("[WARN] Generated Master token: %s", nomadToken)
 		}
-		nomadToken = aclbootstrap.SecretID
-		t.Logf("[WARN] Generated Master token: %s", nomadToken)
-		policy := &nomadapi.ACLPolicy{
-			Name:        "test",
-			Description: "test",
-			Rules: `namespace "default" {
+
+		nomadAuthConfig := nomadapi.DefaultConfig()
+		nomadAuthConfig.Address = nomad.Address()
+		if manuallyBootstrap {
+			nomadAuthConfig.SecretID = nomadToken
+
+			nomadAuth, err := nomadapi.NewClient(nomadAuthConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			err = preprePolicies(nomadAuth)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+		}
+
+		u, _ := docker.NewServiceURLParse(nomadapiConfig.Address)
+		return &Config{
+			ServiceURL: *u,
+			Token:      nomadToken,
+		}, nil
+	})
+	if err != nil {
+		t.Fatalf("Could not start docker Nomad: %s", err)
+	}
+
+	// the container may not be ready yet ... wait a bit
+	time.Sleep(time.Second * 15)
```

Comment:
Why did this become necessary?

Reply:
The container wasn't ready to accept connections, but I now see that `NewServiceRunner` will retry until we get a non-error, so I added a call to get the nomad leader in the setup